src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -359,6 +360,10 @@ private function createFirewall(ContainerBuilder $container, $id, $firewall, &$a
         // Access listener
         $listeners[] = new Reference('security.access_listener');
 
+        if (null !== $logoutListenerId) {
```

Comment:
Can't we just move the whole block that starts with `if (isset($firewall['logout'])) {` here?

Reply:
Nope because factories rely on the definition being in the container so we must create it before but still register last.